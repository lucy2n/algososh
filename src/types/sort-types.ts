export enum SortTypes {
    Bubble = "bubble",
    Select = "select",
}

export enum SortOrderType {
    Increase,
    Decrease
}