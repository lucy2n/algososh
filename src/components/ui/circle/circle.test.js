import { Circle } from "./circle";
import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ElementStates } from "../../../types/element-states";

it('The circle without letter is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The circle with letter is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle letter="T"/>)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The circle with head is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle head='head' />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The circle with react-element in head is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle head={<Circle />} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The circle with tail is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle tail='tail' />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The circle with react-element in tail is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle tail={<Circle />} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The circle with index is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle index={1} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The circle with prop isSmall is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle isSmall={true} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The default circle is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle state={ElementStates.Default} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The changing circle is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle state={ElementStates.Changing} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('The modified circle is rendered without errors', () => {
    const circleWithoutLetter = renderer.create(<Circle state={ElementStates.Modified} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 