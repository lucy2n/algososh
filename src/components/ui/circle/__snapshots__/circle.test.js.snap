// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The changing circle is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`The circle with head is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`The circle with index is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`The circle with letter is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`The circle with prop isSmall is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`The circle with react-element in head is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`The circle with react-element in tail is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`The circle with tail is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`The circle without letter is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`The default circle is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`The modified circle is rendered without errors 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
